Work on clean clang-tidy

loganharbour opened the pull request.

moosebuild commented:
Job [Clang tidy](https://civet.inl.gov/job/1385347/) on 8d3cdeb : invalidated by @loganharbour

loganharbour committed:
Add clang_tidy make target

refs #23614

loganharbour committed:
WIP clang-tidy config

loganharbour committed:
Remove unnecessary vituals

This also resolves a clang-analyzer-optin.cplusplus.VirtualCall
clang-tidy error, in which we're calling a virtual in a constructor

refs #23614

loganharbour committed:
Initialize variables

refs #23614

loganharbour committed:
Remove unread value

refs #23614

loganharbour committed:
Remove unread value

refs #23614

loganharbour committed:
fixup! 249c0aaf767eb7823b12ae8a35316077eec0cc54

loganharbour committed:
fixup! 249c0aaf767eb7823b12ae8a35316077eec0cc54

loganharbour committed:
Initialize values

refs #23614

loganharbour committed:
Only obtain name for non-null object

refs #23614

loganharbour committed:
Remove possible null object query

refs #23614

loganharbour committed:
Move resize to ensure non-null ptr

refs #23614

loganharbour committed:
Add size option to VPP declaration

loganharbour committed:
Initialize tid to invalid value

refs #23614

loganharbour committed:
Use smart pointer for side storage

refs #23614

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1389936/) on 8971d3c wanted to post the following:

The following commit(s) contain fixups:

- 5338bc4aa9: fixup! 249c0aaf767eb7823b12ae8a35316077eec0cc54
- a49d3e5242: fixup! 249c0aaf767eb7823b12ae8a35316077eec0cc54

lindsayad commented:
Feel free to mention me when this is ready for review

loganharbour commented:
Damn... I should have drafted
